dbcon/joblist/tuplehashjoin.cpp

```diff
@@ -373,14 +316,38 @@ void TupleHashJoinStep::startSmallRunners(uint index)
       extendedInfo += oss.str();
     }
     if (!cancelled())
-      joiner->doneInserting();
+      joiners[index]->doneInserting();
   }
 
   boost::mutex::scoped_lock lk(*fStatsMutexPtr);
   fExtendedInfo += extendedInfo;
   formatMiniStats(index);
 }
 
+void TupleHashJoinStep::outOfMemoryHandler(std::shared_ptr<joiner::TupleJoiner> joiner)
+{
+  boost::unique_lock<boost::mutex> sl(saneErrMsg);
+
+  if (cancelled())
+    return;
+  if (!allowDJS || isDML || (fSessionId & 0x80000000) || (tableOid() < 3000 && tableOid() >= 1000))
+  {
+    joinIsTooBig = true;
+    ostringstream oss;
+    oss << "(" << __LINE__ << ") " << logging::IDBErrorInfo::instance()->errorMsg(logging::ERR_JOIN_TOO_BIG);
+    fLogger->logMessage(logging::LOG_TYPE_INFO, oss.str());
+    errorMessage(oss.str());
+    status(logging::ERR_JOIN_TOO_BIG);
+    cout << "Join is too big, raise the UM join limit for now (small runner)" << endl;
+    abort();
```

Comment:
Abort should be explained with the all team

Reply:
Could you elaborate? The same code block had existed for a long time in the THJS. I just moved it into a separate function.